src/components/Database/DatabaseContainer.tsx

```diff
@@ -29,26 +32,62 @@ export const DatabaseContainer: React.FC<Props> = ({
   navigation,
   databases,
   children,
-  fetchDatabases,
+  databasesSubscribe,
+  databasesUnsubscribe,
 }) => {
-  useEffect(() => fetchDatabases(), [fetchDatabases]);
+  useEffect(() => {
+    databasesSubscribe();
+    return databasesUnsubscribe; // Unsubscribe when unmounting.
+  }, [databasesSubscribe, databasesUnsubscribe]);
+  const [dbs, setDbs] = useState(databases);
+
+  let hasNewDbs = false;
+  if (databases !== dbs) {
+    if (
+      dbs === undefined ||
+      databases === undefined ||
+      dbs.join('|') === databases.join('|')
```

Comment:
Are they always sorted? if not this could fail

if so, maybe call the var sortedDbs

Reply:
Or add a comment that they are sorted :)